### ISSUE TITLE
submodule/.gitignore not followed

### ISSUE THREAD

**Issue by cben** [ggreer/the_silver_searcher#1087]
I have a git checkout with several git submodules.  ag searches into the submodules, which is generally a good thing, but it doesn't respect the submodules own .gitignore files, which both pollutes the output and makes it slow.

(in my case, one of the submodules is rbenv, and `rbenv/versions/` contains installations of several ruby versions and all gems I've installed, searching all those is *really* slow...
Interestingly, strace shows `ag` did read `./rbenv/.gitignore`!  `rbenv/.gitignore` does contains `versions`.  But I'm still getting results under rbenv/versions/.)

**Comment by steveprentice** [comment]
Additionally, if a file is ignored by the main git repo, it is ignored by ag in all submodules as well.

```
.gitignore
pom.xml
submodule1/pom.xml
```

Where `.gitignore` contains just `pom.xml`, `ag` will ignore the pom.xml in submodule1. This was unexpected because git itself does not ignore the submodule's pom.xml file.

**Comment by krtek4** [comment]
I agree that it would be really great if `ag` could use the sames rules as git regarding the `.gitignore` files
